src/BlockTypes/ProductGalleryThumbnails.php

```diff
@@ -37,7 +37,48 @@ protected function get_block_type_style() {
 	 * @return string[]
 	 */
 	protected function get_block_type_uses_context() {
-		return [ 'productGalleryClientId', 'postId', 'thumbnailsNumberOfThumbnails', 'thumbnailsPosition' ];
+		return [ 'productGalleryClientId', 'postId', 'thumbnailsNumberOfThumbnails', 'thumbnailsPosition', 'mode' ];
+	}
+
+	/**
+	 * Generate the View All markup.
+	 *
+	 * @param int $remaining_thumbnails_count The number of thumbnails that are not displayed.
+	 *
+	 * @return string
+	 */
+	protected function generate_view_all_html( $remaining_thumbnails_count ) {
+		$view_all_html = '<div class="wc-block-product-gallery-thumbnails__thumbnail__overlay wc-block-product-gallery-dialog-on-click" data-wc-on--click="actions.woocommerce.handleClick">'
+			. '<span class="wc-block-product-gallery-thumbnails__thumbnail__remaining-thumbnails-count wc-block-product-gallery-dialog-on-click">+%1$s</span>'
+			. '<span class="wc-block-product-gallery-thumbnails__thumbnail__view-all wc-block-product-gallery-dialog-on-click">%2$s</span>'
+			. '</div>';
+
+		return sprintf(
+			$view_all_html,
+			esc_html( $remaining_thumbnails_count ),
+			__( 'View all', 'woo-gutenberg-products-block' )
+		);
+	}
```

Comment:
Is there a specific reason for this concatenation? If not, I think that avoiding the concatenation improves the readability of the code. What do you think?

Reply:
Fair point. Removed ✅